streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -2454,4 +2477,36 @@ object ZStreamSpec extends ZIOBaseSpec {
         }
       )
     ) @@ TestAspect.timed
+
+  trait ChunkCoordination[A] {
```

Comment:
I love this!!

Reply:
We can circle back to this when someone actually needs it, but I can see this being useful in some non-test-uses :-)